### ISSUE TITLE
Add support for Projects

### ISSUE THREAD

**Issue by sjparkinson** [Financial-Times/github#26]
Usage as follows...

```
GitHub Projects

Usage:
  github projects:create (--repo|--org|--user) <thing> <name> [<description>]
  github projects:close (--repo|--org|--user) <thing> <name>
```

```
$ github projects:create --repo sjparkinson/example "Example project"
https://github.com/sjparkinson/example/projects/1

$ github projects:create --org Financial-Times "Example project" "This is a description."
https://github.com/orgs/Financial-Times/projects/1

$ github projects:create --user sjparkinson "Example project"
https://github.com/users/sjparkinson/projects/1
```

See https://developer.github.com/v3/projects/ and https://octokit.github.io/rest.js/#api-Projects.

We know that the API is currently not able to configure automation on columns.

----

#### Implementation

- [x] Supports creating a project for an organization
- [x] Supports creating a project for a repository
- [x] Supports creating a project for a user
- [x] Supports creating a project with and without a description
- [ ] Supports closing a project
- [ ] Add tests for `src/lib/github/create-project.js` ?
- ~~[ ] Update `examples/examples.js`~~

**Comment by sjparkinson** [comment]
We should also consider supporting changes to project permissions, https://developer.github.com/v3/projects/#update-a-project.

```
As a developer
I would like to set my project boards permission from the command line
So that I can automate making a working project board on GitHub
```

**Comment by ker-an** [comment]
Something I'd like to discuss is point `Update examples/examples.js`. I'm having a hard time understanding who this directory benefits. Who are these examples for? Because now, it's a CLI tool so a user won't need to refer to this directory.

**Comment by ker-an** [comment]
In person convo with Sam:

- It doesn't make sense to have an examples directory
- `github` is a command line first and we should document that well
- It's going to change rapidly while we refactor the project so maintenance is going to be tough (library will be out of date)
- It doesn't really benefit our users
- Let's delete it!

@simonplend any thoughts?

**Comment by simonplend** [comment]
@ker-an Yup, I'm with you - get rid of the examples! :+1:

**Comment by ker-an** [comment]
For clarity on `Supports closing a project`:

In order to close a project, we need a project ID which as we learned isn't very easy. It's not as simple as getting it from the URL. Sam and I discussed this approach:

1. Extract the project type (e.g. repos, orgs, users) and name (e.g. financial-times-sandbox) from the URL passed in through the command line.

2. Use the GitHub API to list all of the open projects under that account

3. Find the project to close and project_id by using the URL originally passed in as it will be somewhere in the GitHub API response.

4. Use the GitHub API and project_id to close the project.

**Comment by ker-an** [comment]
I did steps 1 to 4 for `orgs` but I hit a blocker with pagination. Organisations with more than 30 projects aren't coming back in the response. I've tried to override the default by setting it to the maximum they allow (100) but that still didn't work.

What's next is to unblock that and apply steps 1 to 4 on `repos` and `users` but that hopefully shouldn't be too much code. It will most likely be laid out like the `commands/create-project.js` file.
